src/Endjin.RecommendedPractices.NuGet/build/Endjin.AutoPackagesLock.targets

```diff
@@ -0,0 +1,65 @@
+      -->
+      <_EndjinExpectedRestorePackagesWithLockFile>true</_EndjinExpectedRestorePackagesWithLockFile>
+      <!--
+        This ensures that during CI builds, the package restore will use exactly what's in the lock
+        file, giving us repeatability. New versions of any of the packages for which we take floating
+        version references (either directly, or implicitly through transitive dependencies) will not
+        be picked up during those builds.
+        When using the project locally, this won't be set, meaning that if any floating references
+        to newer versions become available, they will automatically be acquired, and there will be a
+        record of this happening in source control: the packages.lock.json file will be updated.
+      -->
+      <_EndjinExpectedCiBuildRestoreLockedMode>true</_EndjinExpectedCiBuildRestoreLockedMode>
+    </PropertyGroup>
+
+    <!--
+      Sadly we can't set the build properties directly, because we can't get ourselves injected
+      into the build process at the right point. This is because the part we want to influence
+      is the NuGet package resolution process. But we *are* a NuGet package, so that process
+      necessarily runs before we get to participate. So all we can do is check to see whether
+      the project is set up the way it should be, and make a recommendation if not.
+    -->
+    <Error
+      Condition="('$(_EndjinAutoPackagesLock)' == 'true') And ('$(RestorePackagesWithLockFile)' != '$(_EndjinExpectedRestorePackagesWithLockFile)')"
+      Text="Properties for package lock file generation are not set correctly, and they should normally be enabled for this project type. You can suppress this message by setting the EndjinDisableAutoPackagesLock build variable to true, but in most cases you should add the following PropertyGroup to your project file:&#x0A;&lt;PropertyGroup&gt;&#x0A;  &lt;RestorePackagesWithLockFile&gt;true&lt;/RestorePackagesWithLockFile&gt;&#x0A;  &lt;RestoreLockedMode Condition=&quot;%24(ContinuousIntegrationBuild) == 'true'&quot;&gt;true&lt;/RestoreLockedMode&gt;&#x0A;&lt;/PropertyGroup&gt;&#x0A;"
+      />
+
+    <Error
+      Condition="('$(_EndjinAutoPackagesLock)' == 'true') And ('$(ContinuousIntegrationBuild)' == 'true') And ('$(RestoreLockedMode)' != '$(_EndjinExpectedCiBuildRestoreLockedMode)')"
+      Text="Properties for package lock file generation are not set correctly, and they should normally be enabled for this project type. You can suppress this message by setting the EndjinDisableAutoPackagesLock build variable to true, but in most cases you should add the following PropertyGroup to your project file:&#x0A;&lt;PropertyGroup&gt;&#x0A;  &lt;RestorePackagesWithLockFile&gt;true&lt;/RestorePackagesWithLockFile&gt;&#x0A;  &lt;RestoreLockedMode Condition=&quot;%24(ContinuousIntegrationBuild) == 'true'&quot;&gt;true&lt;/RestoreLockedMode&gt;&#x0A;&lt;/PropertyGroup&gt;&#x0A;"
+      />
```

Comment:
Can the error message text live in a custom property to remove the duplication here?

Reply:
Yes, but then I realised I could just reduce it to a single `Error` task by combining the conditions. So I have removed the duplication that way instead.